### PR TITLE
Add file associations support documentation

### LINKED ISSUE #773: Desktop: Add support for file associations
## Feature Request

The current desktop packaging tasks do not include options for setting up [file associations](https://docs.oracle.com/en/java/javase/16/jpackage/support-application-features.html#GUID-8668A806-8A80-435F-970F-7B2BF65863E4), a particularly valuable feature for desktop apps. 

## Possible Workaround

The `AbstractJPackageTask` allows us to add custom args with the `freeArgs` property, which get passed-through directly to `jpackage`. So to add our file associations, we should be able to follow the documentation linked above and manually add the args we need.

```kotlin
// build.gradle.kts
plugins {
    kotlin("multiplatform")
    id("org.jetbrains.compose")
}
compose.desktop {
    application {
		...
    }
}

tasks.withType<AbstractJPackageTask>().all {
    // add custom args to jpackage to set up file associations
    freeArgs.add("--file-associations")
    freeArgs.add(project.projectDir.resolve("ext.properties"))
}
```

## macOS Problems

The above workaround does not seem to work on macOS (I have not tried it on other OSs). macOS handles file associations differently than Windows or Linux, and has a particularly verbose "registration" process that involves adding a bunch of additional information to the package's `Info.plist` file. Normally, `jpackage` would add all this info to the `Info.plist` file itself in accordance with the `--file-associations` property, but the Compose for Desktop Gradle plugin is [writing that file itself](https://github.com/JetBrains/compose-jb/blob/master/gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/internal/InfoPlistBuilder.kt). This means that even if we add those flags manually using the `freeArgs` property, the application will still not be registered to open custom file types, because the Compose Gradle plugin is not adding the necessary properties to the plist it writes.

In addition, the CfD plugin adds `freeArgs` before it configures itself, and it specifies `--resource-dir` for its own use when writing the `Info.plist`, so we cannot [provide our own resource dir](https://docs.oracle.com/en/java/javase/16/jpackage/override-jpackage-resources.html#GUID-1B718F8B-B68D-4D46-B1DB-003D7729AAB6) to write the plist by hand ourselves either.

## Solution

Rather than writing the `Info.plist` file within the CfD Gradle plugin, it should let `jpackage` do it. 

## Potential Issues

There may already be info CfD is manually adding to `Info.plist` that `jpackage` is not adding (which may have been the reason for manually writing it in the first place). Removing the step to manually write the plist may cause issues with some of the existing Gradle plugin properties, and deeper research into customizing `jpackage` usage or other workarounds may need to be added to fully support them.

### PR DESCRIPTION
**Pull request by zhelenskiy**
Docs for file associations support (https://github.com/JetBrains/compose-multiplatform/pull/4957) which fixes #773.

### CODE REVIEW

**Review by igordmn — COMMENTED** [review]
LGTM, will send to copyediting after approve from Zamula

File: tutorials/Native_distributions_and_local_execution/README.md
Comment (zamulla) [inline tutorials/Native_distributions_and_local_execution/README.md:549]:
```suggestion
            // OS-agnostic file association for .kot files, with icons for all three operating systems.
            fileAssociation(
```

File: tutorials/Native_distributions_and_local_execution/README.md
Comment (zamulla) [inline tutorials/Native_distributions_and_local_execution/README.md:558]:
```suggestion
            // OS-agnostic file association for .kottt files, without icons.
            fileAssociation(
```

File: tutorials/Native_distributions_and_local_execution/README.md
Comment (zamulla) [inline tutorials/Native_distributions_and_local_execution/README.md:522]:
```suggestion
while for OS-specific there is a shorthand with a single `iconFile` parameter.
```

**Review by zamulla — CHANGES_REQUESTED** [review]
Most importantly I think we would benefits from comments in the code, other suggestions are phrasing polish.

**Review by zamulla — APPROVED** [review]
Thanks, looks good 👍

**Review by igordmn — APPROVED** [review]

**Pull request merged**
